PizzaDesignPattren/PizzaDesignPattren/Program.cs

```diff
@@ -0,0 +1,230 @@
+            return final;
+        }
+    }
+    public class ThinPizza
+    {
+        private List<object> _pizza = new List<object>();
+
+        public void PreparePizza(string pizza)
+        { //determine type of dough ,souce,topping
+            this._pizza.Add(pizza);
+        }
+        public string PizzaOrder()
+        {
+            string str = "";
+            for (int i = 0; i < this._pizza.Count; i++)
+            {
+                str += this._pizza[i] + "\n";
+            }
+            return "Thin Pizza order : \n" + str;
+        }
+    }
+
+    public class ThickPizza
+    {
+        private List<string> _pizza = new List<string>();
+
+        public void PreparePizza(string pizza)
+        { //determine type of dough ,souce,topping
+            this._pizza.Add(pizza);
+        }
+        public string PizzaOrder()
+        {
+            string str = "";
+            for (int i = 0; i < this._pizza.Count; i++)
+            {
+                str += this._pizza[i] + "\n";
+            }
+            return "Thick Pizza order : \n" + str;
+        }
+
+    }
+
+    public class Director
+    {
+        private IPizzaBuilder _pizzaBuilder;
+        public IPizzaBuilder PizzaBuilder
+        {
+            set { this._pizzaBuilder = value; }
+        }
+        public void makePizza()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildSauce();
+            this._pizzaBuilder.BuildTopping();
+        }
+        public void makePizzaWithoutSauce()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildTopping();
+        }
+    }
+
+
+
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Director director = new Director();
+            ThickPizzaBuilder thick = new ThickPizzaBuilder();
+            ThinPizzeBuilder thin = new ThinPizzeBuilder();
+            ThinPizzeBuilder thin2 = new ThinPizzeBuilder();
+
+
+            director.PizzaBuilder = thin;
+            // director make order of pizza with souce Ranch and topping cheese
+            director.makePizza();
+            Console.WriteLine(thin.GetPizza().PizzaOrder());
+
+
+            director.PizzaBuilder = thick;
+            // director make another order of pizza with souce Ranch and topping cheese
+            director.makePizza();
+            Console.WriteLine(thick.GetPizza().PizzaOrder());
+
+            director.PizzaBuilder = thin2;
+            //director make another order without souce
+            director.makePizzaWithoutSauce();
+            Console.WriteLine(thin2.GetPizza().PizzaOrder());
+
+            //Bouns 1 make pizza order without director 
+            ThinPizzeBuilder pizza = new ThinPizzeBuilder();
+            pizza.BuildDough();
+            pizza.BuildSauce();
+            pizza.BuildTopping();
+            Console.WriteLine(pizza.GetPizza().PizzaOrder());
+
+
+
+            /*  //Bouns 2 user enter his order !
```

Comment:
Good Job on doing the second  bonus, but I couldn't run the code because it needs some modification on the classes to take the input from the user!